src/subdomains/generic/user/models/user-data/user-data.service.ts

```diff
@@ -163,6 +164,15 @@ export class UserDataService {
       if (identCompleted && pendingVideo) userData.cancelStep(pendingVideo);
     }
 
+    // If KYC level >= 50 and DFX-approval not complete, complete it.
+    if (userData.kycLevel >= KycLevel.LEVEL_50) {
+      const dfxApprovalCompleted = userData.hasCompletedStep(KycStepName.DFX_APPROVAL);
+      if (!dfxApprovalCompleted) {
+        const pendingDfxApproval = userData.getPendingStepWith(KycStepName.DFX_APPROVAL);
+        if (pendingDfxApproval) userData.completeStep(pendingDfxApproval);
```

Comment:
`getPendingStepWith` will not find the step, because it's in `INTERNAL_REVIEW` and not `IN_PROGRESS`